sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -492,6 +508,11 @@ def split_next(self):
                 self._compute_best_split_and_push(right_child_node)
             self.total_find_split_time += time() - tic
 
+        # node.histograms is no longer needed: let's recycle it for memory
+        # efficiency.
+        self.histogram_pool.release(node.histograms)
+        del node.histograms
```

Comment:
The del is there just to avoid accessing this attribute in future code changes by mistake. It would now have a reference to the histogram of another node.

> If we want to explicitly mention that node.histograms is now undefined, maybe we should set it to None instead?

+1 for setting it to `None` explicitly then.